### DISCUSSION TITLE
What is the difference between permutationa and reshape?

### DISCUSSION THREAD

**Discussion by Endtroducing1993** [mrdbourke/pytorch-deep-learning#544] [Q&A]
Hi Team,

Suppose we have a tensor called **tensor_A** of shape (3,4).

What is the difference between **tensor.reshape(4,3)** and **tensor.permute(1,0)**

I have seen by creating a random tensor that we obtain different results.

Can you please suggest, which one to use and where?

**original tensor**
`tensor_original=torch.rand(size=(3,4))`
tensor([[0.9829, 0.2585, 0.1642, 0.6212],
        [0.6378, 0.7740, 0.8801, 0.7784],
        [0.0042, 0.5443, 0.8029, 0.4538]])
        
**rehshaped_tensor**
`tensor_reshape=tensor_original.reshape(4,3)`
tensor([[0.9829, 0.2585, 0.1642],
        [0.6212, 0.6378, 0.7740],
        [0.8801, 0.7784, 0.0042],
        [0.5443, 0.8029, 0.4538]])

**permuted tensor**
`tensor_permuted=tensor_original.permute(1,0)`
tensor([[0.9829, 0.6378, 0.0042],
        [0.2585, 0.7740, 0.5443],
        [0.1642, 0.8801, 0.8029],
        [0.6212, 0.7784, 0.4538]])

**Comment by Perian-Yan** [comment]
Hi,
`reshape` changes the shape of the tensor arbitrarily, as long as the number of entries is the same. It's like assigning an index for each entry: 
In the shape (12,1), indices are [0], [1], ..., [11];
in the shape (4, 3), indices are [0,0], [0,1], [0,2], [1,0], [1,1], ... 

`permute` changes the order of axes. For 2d tensor, it's transpose (you swap x and y axis).
